src/ci/docker/host-x86_64/dist-aarch64-linux/Dockerfile

```diff
@@ -21,6 +21,10 @@ RUN sh /scripts/cross-apt-packages.sh
 COPY scripts/make3.sh /scripts/
 RUN sh /scripts/make3.sh
 
+# We need an aarch64 build of zlib for parity with x86_64.
+# See https://github.com/rust-lang/rust/pull/95545.
+RUN dpkg --add-architecture arm64 && apt-get update && apt-get install zlib1g-dev:arm64
```

Comment:
I suspect that maybe ubuntu 16.04 is too old for arm64 packages to exist?

```
2022-04-10T14:50:13.8693747Z E: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/xenial/main/binary-arm64/Packages  404  Not Found
2022-04-10T14:50:13.8695825Z E: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/xenial-updates/main/binary-arm64/Packages  404  Not Found
2022-04-10T14:50:13.8697689Z E: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/xenial-backports/main/binary-arm64/Packages  404  Not Found
2022-04-10T14:50:13.8699323Z E: Failed to fetch http://security.ubuntu.com/ubuntu/dists/xenial-security/main/binary-arm64/Packages  404  Not Found
```

We probably don't want to upgrade the baseline environment here if we can avoid it (since that bumps libc versions etc), so it might be necessary to either build the library here from scratch (like we do for make etc)... We can also try fetching it from a newer ubuntu (like we do with the cacerts) but that seems potentially annoying due to the libc issues.

Reply:
Hmm, that's a good point. @joshtriplett given you initially suggested `dpkg --add-architecture`, do you have thoughts on this?

Reply:
Ah, I think the problem is that in 16.04 arm64 was on `ports.ubuntu.com`, which requires adding a separate source.

Reply:
You need to modify the existing sources entries to use `deb [arch=amd64,i386]` and add a new one for `ports.ubuntu.com` that uses `deb [arch=arm64]`.